### PR TITLE
Backport https://github.com/apache/incubator-beam/pull/1513

### PR DESCRIPTION
**Pull request by sammcveety**

### CODE REVIEW

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java [hunk 1]
```diff
@@ -754,12 +788,9 @@ public void populateDisplayData(DisplayData.Builder builder) {
         super.populateDisplayData(builder);
         TableReference table = getTable();
```
Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java]: in the Beam PR you deleted this line, too.

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java [hunk 2]
```diff
@@ -1757,6 +1800,28 @@ public Bound to(
...
+      public Bound to(ValueProvider<String> tableSpec) {
+        return toTableRef(NestedValueProvider.of(tableSpec, new TableSpecToTableRef()));
+      }
+
+      /**
+       * Returns a copy of this write transformation, but writing to the specified table.
+       *
+       * <p>Does not modify this object.
+       */
+      private Bound toTableRef(ValueProvider<TableReference> table) {
```
Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java]: I don't understand this change. It's just the VP version of `to(TableReference table)`, right? why change the name.

Comment (dhalperi) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java]: Ben has reminded me why this is needed. I heart java.

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/io/BigQueryIOTest.java [hunk 3]
```diff
@@ -1658,6 +1661,61 @@ public void testRemoveTemporaryTables() throws Exception {
...
+
+  @Test
+  public void testRuntimeOptionsNotCalledInApplyInputQuery() {
+    RuntimeTestOptions options = PipelineOptionsFactory.as(RuntimeTestOptions.class);
+    BigQueryOptions bqOptions = options.as(BigQueryOptions.class);
+    bqOptions.setTempLocation("gs://testbucket/testdir");
+    Pipeline pipeline = TestPipeline.create(options);
+    BigQueryIO.Read.Bound read = BigQueryIO.Read.fromQuery(
+        options.getInputQuery()).withoutValidation();
+    BigQueryIO.Write.Bound write =BigQueryIO.Write
```
Comment (dhalperi) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/io/BigQueryIOTest.java]: fix spacing around =

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/io/BigQueryIOTest.java [hunk 4]
```diff
@@ -1658,6 +1661,61 @@ public void testRemoveTemporaryTables() throws Exception {
...
+
+  @Test
+  public void testRuntimeOptionsNotCalledInApplyInputTable() {
+    RuntimeTestOptions options = PipelineOptionsFactory.as(RuntimeTestOptions.class);
+    BigQueryOptions bqOptions = options.as(BigQueryOptions.class);
+    bqOptions.setTempLocation("gs://testbucket/testdir");
+    Pipeline pipeline = TestPipeline.create(options);
+    BigQueryIO.Read.Bound read = BigQueryIO.Read.from(
+        options.getInputTable()).withoutValidation();
+    BigQueryIO.Write.Bound write =BigQueryIO.Write
```
Comment (dhalperi) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/io/BigQueryIOTest.java]: fix spacing around =

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/io/BigQueryIOTest.java [hunk 5]
```diff
@@ -1658,6 +1661,61 @@ public void testRemoveTemporaryTables() throws Exception {
...
+        .to(options.getOutputTable())
+        .withSchema(NestedValueProvider.of(
+            options.getOutputSchema(), new JsonSchemaToTableSchema()))
+        .withoutValidation();
+    pipeline
+        .apply(read)
+        .apply(write);
+    DisplayData.from(read);
```
Comment (dhalperi) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/io/BigQueryIOTest.java]: add a comment before these 2 lines that we're testing this does not throw.

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/io/BigQueryIOTest.java (hunk 3 above)
Comment (dhalperi) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/io/BigQueryIOTest.java]: Is `write` different the second time around? I don't see anything.

Comment (dhalperi) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/io/BigQueryIOTest.java]: Suggest farming write into a third test. You can use `Create.<TableRow>of()` as an empty input to simplify.

**Comment by sammcveety** [comment]
Done

**Comment by dhalperi** [comment]
LGTM, waiting for green. Please port test improvements back to Beam.

**Pull request merged**